### ISSUE TITLE
SMAPI on the latest Android version works now?

### ISSUE THREAD

**Issue by ElvisTekencio01** [ZaneYork/SMAPI-Android-Installer#365]
So far idk what im doing wrong, dl the latest version but still not working, i have Android 12

**Comment by dmitriyromanyuk** [comment]
On Android 13, everything works more or less even after blocking the Android / data folder

**Comment by ElvisTekencio01** [comment]
Welp itst nice it works on 13 but i have 12, the most recent release won't work.

**Comment by DKqwerty** [comment]
Same. On 12 (and 10), I keep popping Java errors during the "Unpacking" stage in Alpha:
`…FileNotFoundException: open failed: EISDIR (Is a directory)`

**Comment by Kitty035** [comment]
Same for me on Android 11

**Comment by DKqwerty** [comment]
0. Get a file manager that can use the `Android/data` folder (CX File Explorer, MT Manager, older versions of Google Files, etc.).
1. Download and install [the Modded APK](https://drive.google.com/file/d/1C04LQ3V6IAIGHA4ssaie15Wu63Y2wqgM).
2. Download and extract [the ZIP file](https://github.com/ZaneYork/SMAPI-Android-Installer/files/11144397/com.zane.stardewvalley.zip) contents.
3. If `Android\data\com.zane.stardewvalley` doesn't exist, run the modded game once (it will crash, but that's okay for now).
4. Copy the `files\smapi-internal` folder that's in the ZIP into `Android\data\com.zane.stardewvalley\files`.
5. If there is a `StardewValley` folder in your main storage with actual saves, allow the game access to the folder and have it transfer any save files that might exist. Once done (and you double-check that the transfer occurred), you can now safely delete the `StardewValley` folder in main storage. (If a `smapi-installer` folder [name might be wrong, but close] also exists in main storage, it can be safely deleted as well.)

Basically, the instructions provided are confusing/wrong for Android 12 & 10 (or maybe just in general), and the installer app isn't really necessary. I currently have several important mods running: Content Patcher, Mod Config Menu, Time Speed, Skip Fishing Minigame, and of course: Hot Pam.

(And big thanks to @ZaneYork for getting this project off the ground for the rest of us!!)

**Comment by ElvisTekencio01** [comment]
> 0. Get a file manager that can use the `Android/data` folder (CX File Explorer, MT Manager, older versions of Google Files, etc.).
> 
>     1. Download and install [the Modded APK](https://drive.google.com/file/d/1C04LQ3V6IAIGHA4ssaie15Wu63Y2wqgM).
> 
>     2. Download and extract [the ZIP file](https://github.com/ZaneYork/SMAPI-Android-Installer/files/11144397/com.zane.stardewvalley.zip) contents.
> 
>     3. If `Android\data\com.zane.stardewvalley` doesn't exist, run the modded game once (it will crash, but that's okay for now).
> 
>     4. Copy the `files\smapi-internal` folder that's in the ZIP into `Android\data\com.zane.stardewvalley\files`.
> 
>     5. If there is a `StardewValley` folder in your main storage with actual saves, allow the game access to the folder and have it transfer any save files that might exist. Once done (and you double-check that the transfer occurred), you can now safely delete the `StardewValley` folder in main storage. (If a `smapi-installer` folder [name might be wrong, but close] also exists in main storage, it can be safely deleted as well.)
> 
> 
> Basically, the instructions provided are confusing/wrong for Android 12 & 10 (or maybe just in general), and the installer app isn't really necessary. I currently have several important mods running: Content Patcher, Mod Config Menu, Time Speed, Skip Fishing Minigame, and of course: Hot Pam.
> 
> (And big thanks to @ZaneYork for getting this project off the ground for the rest of us!!)

Unfortunately for me it doesn't work, i uninstalled the play store version, followed your instructions and still got an error and the game crashes. What I'm doing wrong? Why destiny won't let me enjoy the game with mods?

**Comment by ElvisTekencio01** [comment]
> 0. Get a file manager that can use the `Android/data` folder (CX File Explorer, MT Manager, older versions of Google Files, etc.).
> 
>     1. Download and install [the Modded APK](https://drive.google.com/file/d/1C04LQ3V6IAIGHA4ssaie15Wu63Y2wqgM).
> 
>     2. Download and extract [the ZIP file](https://github.com/ZaneYork/SMAPI-Android-Installer/files/11144397/com.zane.stardewvalley.zip) contents.
> 
>     3. If `Android\data\com.zane.stardewvalley` doesn't exist, run the modded game once (it will crash, but that's okay for now).
> 
>     4. Copy the `files\smapi-internal` folder that's in the ZIP into `Android\data\com.zane.stardewvalley\files`.
> 
>     5. If there is a `StardewValley` folder in your main storage with actual saves, allow the game access to the folder and have it transfer any save files that might exist. Once done (and you double-check that the transfer occurred), you can now safely delete the `StardewValley` folder in main storage. (If a `smapi-installer` folder [name might be wrong, but close] also exists in main storage, it can be safely deleted as well.)
> 
> 
> Basically, the instructions provided are confusing/wrong for Android 12 & 10 (or maybe just in general), and the installer app isn't really necessary. I currently have several important mods running: Content Patcher, Mod Config Menu, Time Speed, Skip Fishing Minigame, and of course: Hot Pam.
> 
> (And big thanks to @ZaneYork for getting this project off the ground for the rest of us!!)

Ok NOW works, i installed everything following the steps, i copied the zip file before opening the game and ran sd maid se, anw now works. Thanks.
Which mods are good to make the game kinda more simpler, easier? Like, i know about a mod that adds the npcs location in the map, that kind of improvement i want, everymod works?

**Comment by DKqwerty** [comment]
> Ok NOW works, i installed everything following the steps, i copied the zip file before opening the game and ran sd maid se, anw now works. Thanks. Which mods are good to make the game kinda more simpler, easier? Like, i know about a mod that adds the npcs location in the map, that kind of improvement i want, everymod works?

We're getting off topic, but I'll answer: I'm only just beginning fall of year 1, so I don't really know much about long-term balance, but for the first few seasons I'd say **Time Speed** and **Skip Fishing Minigame** are extreme QOL mods.

Fishing is ridiculously difficult—at least for me—such that I've caught zero fish except those few that stay at the bottom of the meter. Just skip it with the mod. You still pay a stamina penalty, plus the waiting and reacting; I think that enough challenge for such an abbreviated day.

With Time Speed, I think IN:1.7/OUT:1.1/MINES:2.2 for "Sec/min" helps give just enough breathing room while still needing to watch the clock. Anyone wanting a bunch of time breathing room (completionists, very casual players, etc.) without completely de-balancing the day cycle should use IN:2.2/OUT:1.7/MINES:3.3+.

**Mod Config Menu** if you don't want to/know how to edit JSON config files.

(Either **Lookup Anything** or **UI Info Suite** would be great, but currently don't work in Android SMAPI. Same with **Skip Intro**.)

**Comment by ElvisTekencio01** [comment]
Ok, thanks, perhaps you know how i can check if all the mods worked? I mean
i saw some but not all, also when the game launched, a screen with text
scrolled and some text was red, but got no errors.

**Comment by DillPickol** [comment]
I'm using a Samsung Galaxy A20 and I can't get it working does anyone know how to on galaxy's?
